src/pkg/packager/composer/list.go

```diff
@@ -0,0 +1,420 @@
+	// the import url and the component name
+	if oci.IsEmptyDescriptor(componentDesc) {
+		h := sha256.New()
+		h.Write([]byte(node.Import.URL + name))
+		id := fmt.Sprintf("%x", h.Sum(nil))
+
+		dir = filepath.Join(cache, "dirs", id)
+
+		message.Debug("creating empty directory for remote component:", filepath.Join("<zarf-cache>", "oci", "dirs", id))
+	} else {
+		tb = filepath.Join(cache, "blobs", "sha256", componentDesc.Digest.Encoded())
+		dir = filepath.Join(cache, "dirs", componentDesc.Digest.Encoded())
+
+		store, err := ocistore.New(cache)
+		if err != nil {
+			return err
+		}
+
+		ctx := context.TODO()
+		// ensure the tarball is in the cache
+		exists, err := store.Exists(ctx, componentDesc)
+		if err != nil {
+			return err
+		} else if !exists {
+			copyOpts := remote.CopyOpts
+			// TODO: investigate why the default FindSuccessors in CopyWithProgress is not working
+			copyOpts.FindSuccessors = content.Successors
+			if err := remote.CopyWithProgress([]ocispec.Descriptor{componentDesc}, store, copyOpts, cache); err != nil {
+				return err
+			}
+			exists, err := store.Exists(ctx, componentDesc)
+			if err != nil {
+				return err
+			} else if !exists {
+				return fmt.Errorf("failed to fetch remote component: %+v", componentDesc)
+			}
+		}
+	}
+
+	if err := utils.CreateDirectory(dir, 0700); err != nil {
+		return err
+	}
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	rel, err := filepath.Rel(cwd, dir)
+	if err != nil {
+		return err
+	}
+	// the tail node is the only node whose relativeToHead is based solely upon cwd<->cache
+	// contrary to the other nodes, which are based upon the previous node
+	ic.tail.relativeToHead = rel
+
+	if oci.IsEmptyDescriptor(componentDesc) {
+		// nothing was fetched, nothing to extract
+		return nil
+	}
+
+	tu := archiver.Tar{
+		OverwriteExisting: true,
+		// removes /<name>/ from the paths
+		StripComponents: 1,
+	}
+	return tu.Unarchive(tb, dir)
+}
+
+// Compose merges the import chain into a single component
+// fixing paths, overriding metadata, etc
+func (ic *ImportChain) Compose() (composed types.ZarfComponent, err error) {
+	composed = ic.tail.ZarfComponent
+
+	if ic.tail.prev == nil {
+		// only had one component in the import chain
+		return composed, nil
+	}
+
+	if err := ic.fetchOCISkeleton(); err != nil {
+		return composed, err
+	}
+
+	// fix paths on the tail
+	fixPaths(&ic.tail.ZarfComponent, ic.tail.relativeToHead)
+	// save mutated tail
+	composed = ic.tail.ZarfComponent
+
+	// dont compose 2x
+	node := ic.tail.prev
+	for node != nil {
+		fixPaths(&node.ZarfComponent, node.relativeToHead)
+
+		// perform overrides here
+		overrideMetadata(&composed, node.ZarfComponent)
+		overrideDeprecated(&composed, node.ZarfComponent)
+		overrideResources(&composed, node.ZarfComponent)
+		overrideActions(&composed, node.ZarfComponent)
+
+		composeExtensions(&composed, node.ZarfComponent, node.relativeToHead)
```

Comment:
```suggestion
	// start with an empty component to compose into
	composed = types.ZarfComponent{}

	// start overriding with the tail node
	node := ic.tail
	for node != nil {
		fixPaths(&node.ZarfComponent, node.relativeToHead)

		// perform overrides here
		overrideMetadata(&composed, node.ZarfComponent)
		overrideDeprecated(&composed, node.ZarfComponent)
		overrideResources(&composed, node.ZarfComponent)
		overrideActions(&composed, node.ZarfComponent)

		composeExtensions(&composed, node.ZarfComponent, node.relativeToHead)

		node = node.prev
	}
```
This logic can be simplified (and is wrong for extensions) - this needs a test.